### PR TITLE
refactor: migrate TransactionSnackbar

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-4345

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Wallet/Dashboard/index.tsx [hunk 1]
```diff
@@ -14,9 +13,7 @@ const dashboard: Plugin.Dashboard.Definition = {
             <>
                 <ConnectWalletDialog />
                 <GasSettingDialog />
-                {getSiteType() !== ExtensionSite.Popup ? (
-                    <TransactionSnackbar pluginID={NetworkPluginID.PLUGIN_EVM} />
-                ) : null}
+                {getSiteType() !== ExtensionSite.Popup ? TransactionSnackbar.open() : null}
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Wallet/Dashboard/index.tsx]: this doesn't look right. open() looks like a side-effect function you should not call it in the render function. if open() does not have a side-effect, please rename that function.

File: packages/mask/src/plugins/Wallet/SNSAdaptor/index.tsx [hunk 2]
```diff
@@ -17,9 +16,9 @@ const sns: Plugin.SNSAdaptor.Definition = {
                 <ApplicationBoardDialog />
                 <ConnectWalletDialog />
                 <GasSettingDialog />
-                {getEnumAsArray(NetworkPluginID).map(({ key, value: pluginID }) => (
-                    <TransactionSnackbar key={key} pluginID={pluginID} />
-                ))}
+                {getEnumAsArray(NetworkPluginID).map(({ key, value: pluginID }) =>
+                    TransactionSnackbar.open({ pluginID }),
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/index.tsx:16]: this doesn't look right. open() looks like a side-effect function you should not call it in the render function. if open() does not have a side-effect, please rename that function.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -30566,6 +30608,7 @@ packages:
       typescript: 4.9.5
       v8-compile-cache-lib: 3.0.1
       yn: 3.1.1
+    dev: true
```
Comment (guanbinrui) [inline pnpm-lock.yaml]: Remove node_modules and run `pnpm install` again.

File: packages/shared/src/modals/TransactionSnackbar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export const TransactionModal = forwardRef<
+    SingletonModalRefCreator<TransactionSnackbarOpenProps>,
+    TransactionSnackbarProps
+>((props, ref) => {
+    const [pluginID, setPluginID] = useState<NetworkPluginID | undefined>()
+
+    const [open, _] = useSingletonModal(ref, {
+        onOpen(props) {
+            setPluginID(props?.pluginID)
```
Comment (guanbinrui) [inline packages/shared/src/modals/TransactionSnackbar/index.tsx]:
```suggestion
            setPluginID(props.pluginID)
```

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**